File: src/main/java/ru/piko/pikopluginlib/PikoPluginData.java
package ru.piko.pikopluginlib;

import org.jetbrains.annotations.NotNull;

import java.io.File;

public class PikoPluginData {

    private final String id;
    private final String name_plugin;
    private PikoPlugin plugin = null; // Сам плагин может быть null
    private @NotNull EStatusPlugin status = EStatusPlugin.BLOCKED; // Статус плагина
    private File file; // Нужен, что бы можно было включить плагин

    public PikoPluginData(String id, PikoPlugin plugin, boolean blocked) {
        this.id = id;
        this.plugin = plugin;
        this.status = blocked ? EStatusPlugin.BLOCKED : EStatusPlugin.ENABLE;
        this.file = plugin.getPluginFile();
        this.name_plugin = plugin.getName();
    }

    public PikoPluginData(String id) {
        this.id = id;
        this.plugin = null;
        this.status = EStatusPlugin.DISABLE;
        this.file = null;
        this.name_plugin = null;
    }

    public void activate(PikoPlugin plugin, boolean blocked) {
        setPlugin(plugin);
        setStatus(blocked ? EStatusPlugin.BLOCKED : EStatusPlugin.ENABLE);
    }

    public void disable() {
        setPlugin(null);
        setStatus(EStatusPlugin.DISABLE);
    }

    public String getNamePlugin() {
        return name_plugin;
    }

    public String getId() {
        return id;
    }

    public PikoPlugin getPlugin() {
        return plugin;
    }

    public void setPlugin(PikoPlugin plugin) {
        this.plugin = plugin;
    }

    public @NotNull EStatusPlugin getStatus() {
        return status;
    }

    public void setStatus(@NotNull EStatusPlugin status) {
        this.status = status;
    }

    public File getFile() {
        return file;
    }

    public void setFile(File file) {
        this.file = file;
    }

}

File: src/main/java/ru/piko/pikopluginlib/Api/PikoPluginData.kt
package ru.piko.pikopluginlib.Api

import java.io.File

class PikoPluginData {
	val id: String
	var namePlugin: String?
		private set
	var countLoad: Int = 0
		private set
	var plugin: PikoPluginAny? = null
		private set
	var status: EStatusPlugin // Статус плагина
		private set
	val file: File? // Нужен, что бы можно было включить плагин
	
	internal constructor(id: String, plugin: PikoPluginAny, blocked: Boolean) {
		this.id = id
		this.plugin = plugin
		this.status = if (blocked) EStatusPlugin.BLOCKED_ENABLE else EStatusPlugin.ENABLE
		this.file = plugin.pluginFile
		this.namePlugin = plugin.name
	}
	
	internal constructor(id: String) {
		this.id = id
		this.plugin = null
		this.status = EStatusPlugin.DISABLE
		this.file = null
		this.namePlugin = null
	}
	
	internal fun addCount() {
		countLoad += 1
	}
	
	fun isFirstLoad(): Boolean = countLoad == 0
	
	fun activate(plugin: PikoPluginAny?, blocked: Boolean) {
		this.plugin = plugin
		status = if (blocked) EStatusPlugin.BLOCKED_ENABLE else EStatusPlugin.ENABLE
		if (namePlugin == null) namePlugin = plugin?.name
	}
	
	fun disable() {
		plugin = null
		status = if (status.isBlocked) EStatusPlugin.BLOCKED_DISABLE else EStatusPlugin.DISABLE
	}
}